### PR TITLE
Migrating CI/CD jobs to macOS 14

### PR DESCRIPTION
**Pull request by huydhn**
We have half the fleet in MacoS 14 already and it has been running fine so far https://github.com/pytorch/pytorch/issues/127490.  So, I'm preparing the final push to replace the rest of them.  This also switches release build from 13 to 14 (GitHub runners)

### CODE REVIEW

**Comment by huydhn** [comment]
A q for @malfet:  Do you plan to keep macOS 13 MPS coverage a bit longer or should we opt for the simpler approach of moving everything to macOS 14 now?

File: .github/workflows/mac-mps.yml [hunk 1]
```diff
@@ -13,29 +13,28 @@ concurrency:
...
-      runner-type: macos-m1-13
+      build-environment: macos-py3-arm64
+      runner-type: macos-m1-stable
       build-generates-artifacts: true
       # To match the one pre-installed in the m1 runners
       python-version: 3.9.12
       test-matrix: |
         { include: [
-          { config: "mps", shard: 1, num_shards: 1, runner: "macos-m1-13" },
           { config: "mps", shard: 1, num_shards: 1, runner: "macos-m2-14" },
```
Comment (huydhn) [inline .github/workflows/mac-mps.yml:28]: This looks like a typo, it should be `macos-m1-14`

**Review by atalman — APPROVED** [review]
lgtm, will need to run validation on nightly macos nightly builds after this one is merged

File: .github/workflows/mac-mps.yml (hunk 1 above)
Comment (huydhn) [reply .github/workflows/mac-mps.yml]: Definitely not a typo, my mistake

**Pull request closed without merging**